serial.c

```diff
@@ -124,7 +124,7 @@ uint8_t serial_read()
     uint8_t data = rx_buffer[tail];
 
     tail++;
-    if (tail == (uint8_t)RX_BUFFER_SIZE) { tail = 0; }
+    if (tail == RX_BUFFER_SIZE) { tail = 0; }
```

Comment:
Why remove this cast? I don't think removing it will hurt anything, but I don't see how that helps, either.